src/strategy-heat-capacitor-functions.js

```diff
@@ -0,0 +1,181 @@
+const { DateTime } = require("luxon");
+//const decreasing_end_prices = require("../test/data/tibber-decreasing-24h.json");
+
+function calculate_opportunities(prices, pattern, amount){
+    //creating a price vector with minute granularity
+    var temp_price = Array(prices.length*60).fill(0)
+    var end_index=[];
+    for(i=0;i<prices.length;i++){
+        temp_price.fill(prices[i],i*60,(i+1)*60);
+        //debugger;
+    }
+    
+    //Calculate weighted pattern
+    var weight = amount/pattern.reduce((a, b) => a + b, 0) //last calculates the sum of all numbers in the pattern
+    var weighted_pattern = pattern.map((x)=>x*weight)
+    
+    //Calculating procurement opportunities. Sliding the pattern over the price vector to find the price for procuring
+    //at time t 
+    dot = (a, b) => a.map((x, i) => a[i] * b[i]).reduce((m, n) => m + n);
```

Comment:
use const when defining constnts like this

Reply:
Done